sdk/identity/identity/src/credentials/clientSecretCredential.ts

```diff
@@ -70,7 +70,7 @@ export class ClientSecretCredential implements TokenCredential {
           grant_type: "client_credentials",
           client_id: this.clientId,
           client_secret: this.clientSecret,
-          scope: typeof scopes === "string" ? scopes : scopes.join(" ")
+          scope: (typeof scopes === "string" ? scopes : scopes.join(" ")).replace(/\/.default$/, "")
```

Comment:
I'll admit, I don't understand what exactly we are doing here (this sort of code looks similar to some of the tricks we played with the `CliCredential` where we had to convert scopes to resources, but I don't think we should be doing this here?  Can you help me understand why this was needed?